Codex: starting post-sync-fix

CI failure class: **code_bug** (code bug — must fix)

```
tegration/test_temporal_backend_parity.py::test_gate_matrix_failure_parity[infra_unavailable_hard_stop:phase1] - AssertionError: runtime phase_statuses for infra_unavailable_hard_stop:phase1 did not match expected policy outcome: {'phase0': 'skipped', 'phase1': 'skipped', 'phase2': 'skipped', 'phase3': 'skipped'}
FAILED tests/integration/test_temporal_backend_parity.py::test_gate_matrix_failure_parity[infra_unavailable_hard_stop:phase2] - AssertionError: runtime phase_statuses for infra_unavailable_hard_stop:phase2 did not match expected policy outcome: {'phase0': 'skipped', 'phase1': 'skipped', 'phase2': 'skipped', 'phase3': 'skipped'}
FAILED tests/integration/test_temporal_backend_parity.py::test_gate_matrix_failure_parity[infra_unavailable_hard_stop:phase3] - AssertionError: runtime phase_statuses for infra_unavailable_hard_stop:phase3 did not match expected policy outcome: {'phase0': 'skipped', 'phase1': 'skipped', 'phase2': 'skipped', 'phase3': 'skipped'}
FAILED tests/integration/test_temporal_backend_parity.py::test_infra_hard_stop_applies_to_phases_parity[infra_unavailable_hard_stop:phase0] - AssertionError: runtime phase_statuses for infra_unavailable_hard_stop:phase0 did not match expected policy outcome: {'phase0': 'skipped', 'phase1': 'skipped', 'phase2': 'skipped', 'phase3': 'skipped'}
FAILED tests/integration/test_temporal_backend_parity.py::test_infra_hard_stop_applies_to_phases_parity[infra_unavailable_hard_stop:phase1] - AssertionError: runtime phase_statuses for infra_unavailable_hard_stop:phase1 did not match expected policy outcome: {'phase0': 'skipped', 'phase1': 'skipped', 'phase2': 'skipped', 'phase3': 'skipped'}
FAILED tests/integration/test_temporal_backend_parity.py::test_infra_hard_stop_applies_to_phases_parity[infra_unavailable_hard_stop:phase2] - AssertionError: runtime phase_statuses for infra_unavailable_hard_stop:phase2 did not match expected policy outcome: {'phase0': 'skipped', 'phase1': 'skipped', 'phase2': 'skipped', 'phase3': 'skipped'}
FAILED tests/integration/test_temporal_backend_parity.py::test_infra_hard_stop_applies_to_phases_parity[infra_unavailable_hard_stop:phase3] - AssertionError: runtime phase_statuses for infra_unavailable_hard_stop:phase3 did not match expected policy outcome: {'phase0': 'skipped', 'phase1': 'skipped', 'phase2': 'skipped', 'phase3': 'skipped'}
FAILED tests/integration/test_temporal_backend_parity.py::test_backend_specific_runtime_ids_are_excluded_from_manifest_parity - AssertionError: unexpected run success for happy: expected True, got False
31 failed, 285 passed, 27 skipped, 3038 warnings in 30.15s
##[error]Process completed with exit code 1.
```

Full CI log: see code_bug run on the PR's checks tab.

--- a/src/orchestrator/definitions.py
+++ b/src/orchestrator/definitions.py
@@ -189,25 +189,53 @@ class _FailClosedDataReadinessResource(ConfigurableResource):
         return _MissingDataReadinessProvider()
 
 
-class _LoadedGatePolicyResource(GatePolicyResource):
-    """Gate policy resource bound to the assembly-time validated profile."""
+# Module-level registry mapping policy_path -> assembly-time
+# GatePolicyProfile snapshot. Kept outside the Pydantic resource class
+# because Pydantic intercepts class-body attributes whose names start with
+# an underscore and turns them into ModelPrivateAttr descriptors, which
+# do not behave as ordinary mutable class attributes.
+_GATE_POLICY_ASSEMBLY_SNAPSHOTS: dict[str, GatePolicyProfile] = {}
 
-    def __init__(self, *, policy_path: str, policy: GatePolicyProfile) -> None:
-        super().__init__(policy_path=policy_path)
-        self._policy = policy
+
+class _LoadedGatePolicyResource(GatePolicyResource):
+    """Gate policy resource bound to the assembly-time validated profile.
+
+    The Dagster Pythonic-config plumbing clones resources at runtime via
+    ``self.__class__(**public_field_values)`` (see
+    ``dagster._config.pythonic_config.resource.ConfigurableResourceFactory.
+    _with_updated_values``). That call only carries Pydantic public fields,
+    so we cannot pass an assembly-time ``GatePolicyProfile`` through a
+    custom ``__init__`` keyword. Instead, we register the validated
+    snapshot under the canonical policy path in
+    ``_GATE_POLICY_ASSEMBLY_SNAPSHOTS`` and look it up from there whenever
+    the resource (or one of its clones) is asked for ``.policy``.
+    ``setup_for_execution`` is overridden to a no-op so the on-disk policy
+    is never silently reloaded after assembly: if the file on disk diverges
+    from the registered snapshot, gate decisions still classify against the
+    snapshot validated by ``build_definitions``.
+    """
+
+    @staticmethod
+    def register_snapshot(
+        policy_path: str,
+        policy: GatePolicyProfile,
+    ) -> None:
+        _GATE_POLICY_ASSEMBLY_SNAPSHOTS[policy_path] = policy
 
     def setup_for_execution(self, context: object) -> None:
         return None
 
     @property
     def policy(self) -> GatePolicyProfile:
-        if self._policy is None:
+        snapshot = _GATE_POLICY_ASSEMBLY_SNAPSHOTS.get(self.policy_path)
+        if snapshot is None:
             msg = (
-                "gate policy snapshot was not initialized during "
+                "gate policy snapshot for "
+                f"{self.policy_path!r} was not initialized during "
                 "Definitions assembly"
             )
             raise RuntimeError(msg)
-        return self._policy
+        return snapshot
 
 
 def build_definitions(
@@ -280,10 +308,8 @@ def build_definitions(
         _FailClosedLLMHealthProbeResource(),
     )
     data_readiness_resource = _data_readiness_resource(resource_bundle.resources)
-    gate_policy_resource = _LoadedGatePolicyResource(
-        policy_path=policy_path_str,
-        policy=policy,
-    )
+    _LoadedGatePolicyResource.register_snapshot(policy_path_str, policy)
+    gate_policy_resource = _LoadedGatePolicyResource(policy_path=policy_path_str)
     resources = _guard_infrastructure_resources(
         {
             "gate_policy": gate_policy_resource,

--- a/tests/integration/test_infra_hard_stop.py
+++ b/tests/integration/test_infra_hard_stop.py
@@ -63,43 +63,31 @@ def test_daily_cycle_hard_stops_on_guarded_resource_init_failure(
     assert f"fake {failing_resource_key} unavailable" in str(payload["summary"])
 
 
-def test_daily_cycle_alerts_when_gate_policy_resource_load_fails(
+def test_build_definitions_fails_fast_when_gate_policy_path_missing(
     dagster_module: object,
-    dagster_instance: object,
     tmp_dbt_project: Path,
     tmp_path: Path,
-    caplog: pytest.LogCaptureFixture,
     monkeypatch: pytest.MonkeyPatch,
 ) -> None:
+    """Missing gate policy must raise loudly during Definitions assembly.
+
+    Per issue #111, the prior behavior (catch the FileNotFoundError, default
+    to dagster_only, register the resource and let it surface later) is a
+    silent backend swap that masks misconfiguration. The contract is now
+    fail-fast: build_definitions must validate the policy up front so the
+    backend selection, sensor wiring, and gate classification all derive
+    from the same validated profile.
+    """
     dagster = dagster_module
     missing_policy_path = tmp_path / "missing_gate_policy.yaml"
-    defs = _build_defs(
-        dagster,
-        monkeypatch=monkeypatch,
-        stub_policy_path=str(missing_policy_path),
-        failing_resource_key="none",
-    )
-    dagster.Definitions.validate_loadable(defs)
 
-    with caplog.at_level(logging.WARNING, logger="orchestrator.alerting.dispatcher"):
-        result = defs.get_job_def("daily_cycle_job").execute_in_process(
-            instance=dagster_instance,
-            raise_on_error=False,
-            tags={"cycle_id": "cycle-20260416"},
+    with pytest.raises(FileNotFoundError, match=missing_policy_path.name):
+        _build_defs(
+            dagster,
+            monkeypatch=monkeypatch,
+            stub_policy_path=str(missing_policy_path),
+            failing_resource_key="none",
         )
-
-    materialized_keys = asset_materialization_keys(result)
-    payload = _single_alert_for_resource(caplog.records, "gate_policy")
-
-    assert result.success is False
-    assert dagster.AssetKey(["graph_promotion"]) not in materialized_keys
-    assert payload["phase"] == "phase0"
-    assert payload["failure_class"] == "infra"
-    assert payload["action"] == "fail_run"
-    assert payload["scenario_id"] == "infra_unavailable_hard_stop"
-    assert payload["runbook_url"] == "docs/RUNBOOK_P5.md#phase2-infra-fail_run"
-    assert payload["failed_node"] == "gate_policy"
-    assert missing_policy_path.name in str(payload["summary"])
 
 
 def test_build_definitions_alerts_when_provider_get_resources_fails(

